### ISSUE TITLE
Feature request: add integro-differential equation support

### ISSUE THREAD

**Issue by sdwfrost** [SciML/ModelingToolkit.jl#572]
There aren't many implementations of integro-differential equations which are useful for many things e.g. distributed delays, non-exponential distributions of passage times, time-since-infection epidemiological models (see [here](https://github.com/rajeshrinet/pyrosstsi/blob/master/docs/TSI_writeup.pdf)) and it would be great if MTK/the DiffEq ecosystem could support these.

**Comment by ChrisRackauckas** [comment]
Yeah, I think the best way to do this is to add it to the PDESystem spec so that way we can utilize it in physics-informed neural networks. @KirillZubov should be in tune with this.
